### PR TITLE
Fixing NullPointerExceptions in PullRequests

### PR DESCRIPTION
**Pull request by 2improveIT**
Fixing NullPointerExceptions

### Your checklist for this pull request

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

**Comment by casz** [comment]
Thanks for resubmitting this PR 😊

**Comment by casz** [comment]
link to see current commit author: https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/245/commits/710faec6e6072648bc641618ca1908bfa40511e1.patch

@2improveIT 
You might want to set up an email alias on your GitHub account 🥇
https://help.github.com/articles/adding-an-email-address-to-your-github-account/

Or look into fixing your git user.email config 😅  
https://help.github.com/articles/setting-your-commit-email-address-in-git
then you would need to amend your commits: https://stackoverflow.com/a/3042512
after changing author

to get proper credit for your commits 🏆

File: src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java [hunk 1]
```diff
@@ -356,11 +356,18 @@ public BitbucketCommit call() throws Exception {
...
+        try {
+            BitbucketServerBranch branch = (BitbucketServerBranch) pr.getSource().getBranch();
+            if (branch != null) {
+                branch.setCommitClosure(new CommitClosure(branch.getRawNode()));
+            }
+            branch = (BitbucketServerBranch) pr.getDestination().getBranch();
+            if (branch != null) {
+                branch.setCommitClosure(new CommitClosure(branch.getRawNode()));
+            }
+        } catch (Exception e) {
```
Comment (darxriggs) [inline src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java]: Do you really want to catch ALL exceptions here?

File: src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java [hunk 2]
```diff
@@ -356,11 +356,18 @@ public BitbucketCommit call() throws Exception {
...
+            LOGGER.log( Level.SEVERE, "setupClosureForPRBranch", e);
```
Comment (darxriggs) [inline src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java]: The formatting could be improved by removing the trailing space after `LOGGER.log( `.

File: src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java (hunk 1 above)
Comment (casz) [reply src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java]: Yup, I missed that 😱

Comment (darxriggs) [reply src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java]: @2improveIT ping

Comment (2improveIT) [reply src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java]: adapted the code

File: src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java (hunk 2 above)
Comment (2improveIT) [reply src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java]: adapted the code

**Pull request merged**